### ISSUE TITLE
Transpose of sparse tensor Variables no longer works in 0.3

### ISSUE THREAD

**Issue by gpleiss** [pytorch/pytorch#4236]
Sparse tensor Variables seem to no longer support the transpose operation.

```python
import torch
i = torch.LongTensor([[0, 1, 1], [2, 0, 2]])
v = torch.FloatTensor([3, 4, 5])
sparse_mat = torch.sparse.FloatTensor(i, v, torch.Size([2,3]))
sparse_mat.t()
```

This code will run. However, if we wrap `sparse_mat` in a variable...

```python
from torch.autograd import Variable
sparse_var = Variable(sparse_mat)
sparse_var.t()
```

This code returns the following error:

`RuntimeError: t is not implemented for type torch.sparse.FloatTensor`

This code was working just fine before upgrading to 0.3.

**Comment by zou3519** [comment]
We should migrate the sparse tensor [cwrap declarations](https://github.com/pytorch/pytorch/blob/master/torch/csrc/generic/methods/SparseTensor.cwrap) to ATen's [declarations.cwrap](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/Declarations.cwrap) for the functions that exist. I can try to do this.

**Comment by zou3519** [comment]
Task tracker for sparse ops on tensor that need to be enabled for Variable:

- [ ] transpose
- [ ] transpose_
- [ ] t
- [ ] t_
- [ ] zero_
- [ ] zeros
- [ ] zeros_like
- [ ] sub
- [ ] sub_
- [ ] mul
- [ ] mul_
- [ ] pow

**Comment by soumith** [comment]
closed via #4687 #4779 #4780 #4785
